src/transport/abstract-transport.js

```diff
@@ -20,6 +20,7 @@ function AbstractTransport(address, timeout, messageHandler) {
   this.awaitingPump = false;
   this.connected = false;
   this.backoffMilliseconds = 1;
+  this.nextReconnectListenerToken = 1;
```

Comment:
> If you are referring to the actual block subscription to the remote Ethereum Node (i.e., `eth_subscribe`) then there will be one per instance of `BlockStream`.  In most cases, the app will probably only have one `BlockStream` for its life and it can be treated as a singleton.  However, the code is written so this isn't strictly necessary (makes testing easier and could be useful one day).

When you instantiate `BlockStream`, this automatically instantiates a `BlockNotifier`, which instantiates either a PollingBlockNotifier or a SubscriptionBlockNotifier, which in turn automatically calls `transport.subscribeToNewHeads` which -- if I'm following this correctly -- is the function defined in ethrpc-transport-adapter that increments `nextToken` and also calls `ethrpc.subscribeNewHeads` (which calls `eth_subscribe` under the hood).  Is this correct, or have I missed something?  If it's correct, then `nextToken` should only get incremented when a new `eth_subscribe` is issued.